lib/api/repository/survey_repository.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:flutter_survey/api/exception/network_exceptions.dart';
+import 'package:flutter_survey/api/survey_service.dart';
+import 'package:flutter_survey/models/survey.dart';
+import 'package:injectable/injectable.dart';
+
+abstract class SurveyRepository {
+  Future<List<Survey>> getSurveys(int pageNumber, int pageSize);
+}
+
+@Singleton(as: SurveyRepository)
+class SurveyRepositoryImpl extends SurveyRepository {
+  SurveyService _surveyService;
+
+  SurveyRepositoryImpl(this._surveyService);
+
+  @override
+  Future<List<Survey>> getSurveys(int pageNumber, int pageSize) async {
+    try {
+      final responses = await _surveyService
+          .getSurveyList(pageNumber, pageSize)
+          .then((response) => response.data
+              .map((apiResponse) => apiResponse.attributes)
```

Comment:
@luongvo  Can we map directly to `Survey`?

Reply:
yep seems like the below mapping is unnecessary making another round of list iteration O(2n):

`responses.map((response) => response.toSurvey()).toList()` 🤔